Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -308,23 +308,78 @@ Add the following to build.gradle:
 ### Testing Exception Handling
 
 #### Overview
-
--- TODO [Give an overview about the functionality and it's potential weaknesses] --
+Exceptions can often occur when an application gets into a non-normal or erroneous state. Both in Java and C++ Exceptions can be thrown when such state occurs. 
+Testing exception handling is about reassuring that the application will handle the exception and get to a safe state without exposing any sensitive information at both the UI and the ADB logs.
 
 #### Static Analysis
 
 Review the source code to understand and identify how the application handles various types of errors (IPC communications, remote services invocation, etc). Here are some examples of the checks to be performed at this stage :
 
 * Verify that the application use a well-designed and unified scheme to handle exceptions<sup>[1]</sup>.
-* Verify that the application doesn't expose sensitive information while handling exceptions, but are still verbose enough to explain the issue to the user.
+* Verify that standard `RuntimeException`s such as `NullPointerException`, `IndexOfBoundsException`, `ActivityNotFoundException`, `CancellationException`, `SQLException`, etc.. are anticipated upon by creating proper null-checks, bound-checks and alike. See <sup>[2]</sup> for an overview of the provided child-classes of `RuntimeException`. If the developer still throws a child of `RuntimeException` then this should always be intentional and that intention should be handled by the calling method.
```

Comment:
Hi! Thanks for the feedback, wll fix the first 2 remarks. According to https://developer.android.com/reference/android/database/SQLException.html SQLException is a RuntimeException :-(...